### PR TITLE
Allow storage of Ed25519 Signatures

### PR DESCRIPTION
**Pull request by matheusd**
Following up on #1536, while Ed25519 keys are internally supported, signatures can't currently be stored due to the use of an Enum in the appropriate field.

This fixes the issue for new installations. Please advise if some form of upgrade script is needed (I couldn't find any place to add them to).

### Checklist

- [ ] I have updated the [CHANGELOG](CHANGELOG.md).
  - Adjust the draft version number according to [semantic versioning](https://semver.org/) rules.
- [ ] I have updated [documentation](docs/) accordingly (including the [feature implementation matrix](docs/Feature_Implementation_Matrix.md)).

### CODE REVIEW

**Comment by matheusd** [comment]
@googlebot I signed it!

**Comment by marcopeereboom** [comment]
This is great. I would love to see this go in.

File: crypto/signatures.go [hunk 1]
```diff
@@ -31,7 +31,7 @@ func SignatureAlgorithm(k gocrypto.PublicKey) sigpb.DigitallySigned_SignatureAlg
 		return sigpb.DigitallySigned_ECDSA
 	case *rsa.PublicKey:
 		return sigpb.DigitallySigned_RSA
-	case *ed25519.PublicKey:
+	case ed25519.PublicKey:
```
Comment (Martin2112) [inline crypto/signatures.go:34]: Why is this different to the other?

Comment (gdbelvin) [reply crypto/signatures.go:34]: The underlying type is `[]byte`. https://golang.org/src/crypto/ed25519/ed25519.go?s=1385:1406#L31

The other types are `struct`s that have pointers. 

This change looks correct

**Comment by Martin2112** [comment]
Looks like the travis failure is the lint issue that Gary is fixing so this is ok to merge.

**Comment by Martin2112** [comment]
@matheusd if you rebase on master and let travis run & pass you should be able to merge.

**Comment by matheusd** [comment]
The only thing failing now is code coverage (which I'm not sure it's applicable since there are no added lines). Do I have to take any other actions?

**Comment by Martin2112** [comment]
Merged. Thanks for the contribution.

**Pull request merged**